### PR TITLE
Move `fixtures.bzl` and `xcodeproj_tests.bzl` to `xcodeproj/internal`

### LINKED ISSUE #163: Move fixtures.bzl to `xcodeproj/internal` and expose its rules/macros in `xcodeproj/testing.bzl`.
Related to #115.

Moving the fixtures files allows the rules to be accessible to child workspaces.

### PR DESCRIPTION
**Pull request by cgrindel**
Closes #163.

- Refactored `xcodeproj_test_suite` to not contain target references.
- Added `xcodeproj_tests.from_fixture` and `xcodeproj_tests.from_fixtures` to aid in the declaration of fixture tests.
- Added a new `fixtures.bzl` that contains a list of fixture related names and targets. These are used by fixture related declarations.

### CODE REVIEW

File: test/fixtures/fixtures.bzl [hunk 1]
```diff
@@ -1,133 +1,12 @@
-"""Functions for updating test fixtures."""
+"""Constants for Fixture Declarations."""
```
Comment (brentleyjones) [inline test/fixtures/fixtures.bzl:1]:
```suggestion
"""Constants for fixture declarations."""
```

File: test/fixtures/fixtures.bzl [hunk 2]
```diff
@@ -1,133 +1,12 @@
...
-)
+FIXTURE_BASENAMES = [
+    "cc",
+    "command_line",
+    "generator",
+    "ios_app",
+]
 
-# Transition
+FIXTURE_PACKAGES = ["//test/fixtures/{}".format(b) for b in FIXTURE_BASENAMES]
```
Comment (brentleyjones) [inline test/fixtures/fixtures.bzl:10]:
```suggestion
_FIXTURE_PACKAGES = ["//test/fixtures/{}".format(b) for b in FIXTURE_BASENAMES]
```

File: test/fixtures/fixtures.bzl [hunk 3]
```diff
@@ -1,133 +1,12 @@
...
-            ["{}.xcodeproj/**/*".format(project_name)],
-            exclude = [
-                "{}.xcodeproj/xcshareddata/**/*".format(project_name),
-                "{}.xcodeproj/**/xcuserdata/**/*".format(project_name),
-                "{}.xcodeproj/*.xcuserdatad/**/*".format(project_name),
-            ],
-        ),
-        visibility = ["//test:__subpackages__"],
-    )
+FIXTURE_TARGETS = ["{}:xcodeproj".format(p) for p in FIXTURE_PACKAGES]
```
Comment (brentleyjones) [inline test/fixtures/fixtures.bzl:12]:
```suggestion
FIXTURE_TARGETS = ["{}:xcodeproj".format(p) for p in _FIXTURE_PACKAGES]
```

File: test/internal/xcodeproj_tests/xcodeproj_tests_tests.bzl [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    ]
+    asserts.equals(env, expected, actual, "Multiple fixture packages.")
+
+    return unittest.end(env)
+
+from_fixtures_test = unittest.make(_from_fixtures_test)
+
+def xcodeproj_tests_test_suite():
+    return unittest.suite(
+        "xcodeproj_tests_tests",
```
Comment (brentleyjones) [inline test/internal/xcodeproj_tests/xcodeproj_tests_tests.bzl:65]:
```suggestion
def xcodeproj_tests_test_suite(name):
    return unittest.suite(
        name,
```

File: test/internal/xcodeproj_tests/BUILD [hunk 5]
```diff
@@ -0,0 +1,3 @@
+load(":xcodeproj_tests_tests.bzl", "xcodeproj_tests_test_suite")
+
+xcodeproj_tests_test_suite()
```
Comment (brentleyjones) [inline test/internal/xcodeproj_tests/BUILD:3]:
```suggestion
xcodeproj_tests_test_suite(name = "xcodeproj_tests_tests")
```

File: xcodeproj/internal/xcodeproj_tests.bzl [hunk 6]
```diff
@@ -1,7 +1,47 @@
...
+        basename = None,
+        expected_spec = None,
+        expected_xcodeproj = None):
+    if target_under_test == None:
+        fail("Need to specify the target under test.")
+    if target_under_test.find(":") < 0:
+        target_under_test += ":xcodeproj"
+
+    test_target_parts = target_under_test.split(":")
+    pkg = test_target_parts[0]
```
Comment (brentleyjones) [inline xcodeproj/internal/xcodeproj_tests.bzl:20]: I don't think we want to default this.

Can we turn it into a label if it's not, then use it's APIs? If not, we can fail if it's not a fully qualified string.

File: xcodeproj/testing.bzl [hunk 7]
```diff
@@ -0,0 +1,19 @@
+"""Testing Related Rules and Macros"""
```
Comment (brentleyjones) [inline xcodeproj/testing.bzl:1]:
```suggestion
"""Testing related rules and macros."""
```

**Review by brentleyjones — COMMENTED** [review]
Thanks for doing this! Just a couple nits/questions.

File: test/internal/xcodeproj_tests/xcodeproj_tests_tests.bzl [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+    # Specify target
+    actual = xcodeproj_tests.from_fixture("//path/to/pkg:custom_xcodeproj")
+    expected = struct(
+        basename = "pkg",
+        target_under_test = "//path/to/pkg:custom_xcodeproj",
+        expected_spec = "//path/to/pkg:spec.json",
+        expected_xcodeproj = "//path/to/pkg:xcodeproj_output",
+    )
+    asserts.equals(env, expected, actual, "Specify fixture target.")
```
Comment (brentleyjones) [inline test/internal/xcodeproj_tests/xcodeproj_tests_tests.bzl:17]:
```suggestion
    asserts.equals(env, expected, actual, "Specifing fixture target")
```

File: test/internal/xcodeproj_tests/xcodeproj_tests_tests.bzl [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+        expected_spec = "//path/to/pkg:custom_spec.json",
+        expected_xcodeproj = "//path/to/pkg:custom_xcodeproj_output",
+    )
+    expected = struct(
+        basename = "custom_basename",
+        target_under_test = "//path/to/pkg:custom_xcodeproj",
+        expected_spec = "//path/to/pkg:custom_spec.json",
+        expected_xcodeproj = "//path/to/pkg:custom_xcodeproj_output",
+    )
+    asserts.equals(env, expected, actual, "Specify everything.")
```
Comment (brentleyjones) [inline test/internal/xcodeproj_tests/xcodeproj_tests_tests.bzl:32]:
```suggestion
    asserts.equals(env, expected, actual, "Specifing everything")
```

File: test/internal/xcodeproj_tests/xcodeproj_tests_tests.bzl [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+def _from_fixtures_test(ctx):
+    env = unittest.begin(ctx)
+
+    packages = ["//foo", "//bar"]
+    actual = xcodeproj_tests.from_fixtures(packages)
+    expected = [
+        xcodeproj_tests.from_fixture("//foo"),
+        xcodeproj_tests.from_fixture("//bar"),
+    ]
+    asserts.equals(env, expected, actual, "Multiple fixture packages.")
```
Comment (brentleyjones) [inline test/internal/xcodeproj_tests/xcodeproj_tests_tests.bzl:47]:
```suggestion
    asserts.equals(env, expected, actual, "Multiple fixture packages")
```

File: xcodeproj/BUILD [hunk 11]
```diff
@@ -0,0 +1 @@
+package(default_visibility = ["//visibility:public"])
```
Comment (brentleyjones) [inline xcodeproj/BUILD:1]: Is this needed?

Comment (cgrindel) [reply xcodeproj/BUILD]: I typically make items in that package public as it is the location to find the publicly available rules, macros, and libraries. I can remove it until/if we need it.

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**